src/plugins/locale/index.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable dot-notation */
+import type { EsDayPlugin, Locale, UnitType } from 'esday'
+import * as C from '~/core/constant'
+import { prettyUnit, undefinedOr } from '~/core/utils'
+import en from '~/locales/en'
+
+const LocaleStore: Map<string, Locale> = new Map()
+
+let $localeGlobal = 'en'
+
+export function getLocale(name: string): Locale {
+  return LocaleStore.get(name) || en
+}
+
+export function registerLocale(locale: Locale, rename?: string): void {
+  LocaleStore.set(rename || locale.name, locale)
+}
+
+declare module 'esday' {
+/*   interface EsDay {
+    $locale: () => Locale
+  } */
+
+  interface EsDay {
+    locale: (localeName: string) => EsDay
+  }
+
+  interface EsDayFactory {
+    /**
+     * use locale as global
+     */
+    locale: (localeName: string) => EsDayFactory
+    /**
+     * register locale
+     */
+    registerLocale: (locale: Locale, rename?: string) => EsDayFactory
+  }
+}
+
+export const localePlugin: EsDayPlugin<{}> = (_, dayClass, dayFactory) => {
+  // @ts-expect-error $locale is private method
+  dayClass.prototype.$locale = function () {
+    // @ts-expect-error $l is private property
+    return getLocale(this['$l'] || 'en')
```

Comment:
I am really no fan of these 1 letter variables - especially (but not only), if they are created dynamically. This naming convention (with some more examples in the core module) is for sure not self explanatory not for contributors nor perhaps for the author in 1 year 😄).

Would you mind using a more 'speaking' name?

Reply:
I agree. There is too much unclear variable names in dayjs. How about `$locale_name` ?